hyperdome_client/threads.py

```
@@ -72,20 +72,232 @@ def run(self):
             return
 
 
-class WebThread(QtCore.QThread):
+class TaskSignals(QtCore.QObject):
+    success = QtCore.pyqtSignal(str)
+    error = QtCore.pyqtSignal(str)
+
+
+class SendMessageTask(QtCore.QRunnable):
     """
-    Starts the web service
+    send client message to server
     """
-    success = QtCore.pyqtSignal()
-    error = QtCore.pyqtSignal(str)
+    signals = TaskSignals()
 
-    def __init__(self, mode):
-        super(WebThread, self).__init__()
-        self.mode = mode
-        self.mode.common.log('WebThread', '__init__')
+    def __init__(self,
+                 server: Server,
+                 session: requests.Session,
+                 uid: str,
+                 message: str):
+        super(SendMessageTask, self).__init__()
+        self.server = server
+        self.session = session
+        self.uid = uid
+        self.message = message
+
+    def run(self):
+        try:
+            send_message(self.server, self.session, self.uid, self.message)
+        except requests.RequestException:
+            self.signals.error.emit("Couldn't send message")
```

You should probably log or something here; as it is, it looks like the exception context is being swallowed. Unless you know what might cause that to happen? In which case maybe add a comment or clarify the "Couldn't send message"

---

All task exceptions are really generic and need to be refined.

Leaving for now.